### ISSUE TITLE
Sass Challenges intermittently report failed tests for correct solutions

### ISSUE THREAD

**Issue by P1xt** [freeCodeCamp/freeCodeCamp#17570]
#### Describe your problem and - if possible - how to reproduce it
The test cases for this challenge do not accept valid formatted code which follows the instructions and example given.

-----

**Note:** The example I give is one specific challenge, but this issue is NOT unique to the challenge, it happens throughout the Sass Challenges.

-----

Frequently in the sass challenges, valid solutions are not accepted. However, submitting the challenge again (making no changes whatsoever) a few times will cause the challenge to pass.

Steps to reproduce:

- Go to [Sass: Use @each to Map Over Items in a List](https://learn.freecodecamp.org/front-end-libraries/sass/use-each-to-map-over-items-in-a-list)
- Solve the challenge, the following is the correct solution:
```
  @each $color in blue, black, red {
    .#{$color}-bg {background-color: $color;}
  }
```
- Submit the solution multiple times

Sometimes, the solution will pass, other times, it will fail. There are no errors to the console, and whether the challenge will accept a correct solution seems completely random. Just press submit a few times and it will finally pass.

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/front-end-libraries/sass/use-each-to-map-over-items-in-a-list

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: Version 66.0.3359.181 (Official Build) (64-bit)
* Operating System: Manjaro Linux 17.1.10

#### If possible, add a screenshot here
![sass_challenge](https://user-images.githubusercontent.com/322175/41370700-f1dcab9c-6f0d-11e8-80b1-270212d07310.png)

![sass_same_challenge_passed](https://user-images.githubusercontent.com/322175/41370712-f86960c2-6f0d-11e8-8dbc-610bb1b76732.png)

**Comment by ipa-1** [comment]
Browser Name: Chrome
Browser Version: Version 67.0.3396.87 (Official Build) (64-bit)
Operating System: Windows 10 Home

https://learn.freecodecamp.org/front-end-libraries/sass/use-each-to-map-over-items-in-a-list

![image](https://user-images.githubusercontent.com/28264369/41422123-44432f9c-6fc6-11e8-9f5b-fb0877a63201.png)

https://learn.freecodecamp.org/front-end-libraries/sass/use-for-to-create-a-sass-loop/

![image](https://user-images.githubusercontent.com/28264369/41422349-bcf1edc0-6fc6-11e8-8c38-7fb15ef80bd3.png)

https://learn.freecodecamp.org/front-end-libraries/sass/store-data-with-sass-variables

![image](https://user-images.githubusercontent.com/28264369/41422438-ebcf30b2-6fc6-11e8-96c3-10d421f377a7.png)

I've tried running the tests many times for each case, but it never randomly passes

**Comment by QuincyLarson** [comment]
@P1xt Thanks for the detailed instructions. I was unable to reproduce this in Chrome, but I did reproduce it in FireFox. 

@Bouncey Any idea what could be causing this?

**Comment by brimarq** [comment]
I had similar issues using Firefox 60.0.2 (64-bit) on macOS 10.12.6. Initial partial failures of correct solutions on some exercises. The results pane behaved appropriately, and I even opened dev tools to check and see that my results were correct. I just waited a few seconds and re-ran the tests on some, or went back to re-run tests after having moved on to others. There doesn't seem to be any pattern as to how many times they need to be re-run. Eventually I got them all to pass, though. Weird.

**Comment by ivan3123708** [comment]
Browser Name: Google Chrome
Browser Version: (64-bit) 67.0.3396.87 (Official Build) (64-bit)
Operating System: Ubuntu 18.04 LTS

I'm having same 3 issues as @ipa-1, tests never pass.

**Comment by mattszcz** [comment]
#12699 root issue

I also have confirmed intermittent test passing (by spamming "Run the tests") on:
https://learn.freecodecamp.org/front-end-libraries/sass/store-data-with-sass-variables
https://learn.freecodecamp.org/front-end-libraries/sass/use-each-to-map-over-items-in-a-list

but not on the other 2 broken Sass lessons

**Comment by JohnnyBizzel** [comment]
I just tried the 1st challenge and got stuck
```
<style type='text/sass'>
  
  $text-color: red;
  
  .header{
    text-align: center;
  }
  .blog-post, h2 {
    color:  $text-color ;
  }
</style>
```
However, I notice the issue is solved by removing extra space like so in the last style declaration:
```
<style type='text/sass'>
  
  $text-color: red;
  
  .header{
    text-align: center;
  }
  .blog-post, h2 {
    color: $text-color;
  }
</style>
```

**Comment by thecodingaviator** [comment]
@P1xt / @JohnnyBizzel I think this should've been fixed now?

**Comment by thecodingaviator** [comment]
@P1xt I just check and ran the same solution 5 times, it passed every time. Please close the issue

**Comment by thecodingaviator** [comment]
@JohnnyBizzel Your issue was fixed in #18828 , @scissorsneedfoodtoo please close this issue.

**Comment by scissorsneedfoodtoo** [comment]
Thanks for confirming this was fixed, @thecodingaviator. @P1xt, I'll close this for now, but it will take some time for the fix to be reflected on the live site. Please feel free to reopen it if you're still having problems with this in the future.
